zdm/tests/mc_statistics.py

```diff
@@ -0,0 +1,386 @@
+
+"""
+This script illustrates how to generate MC samples and evaluate them
+Note that in this case, only MC samples for:
+- one survey (ASKAP fly's eye)
+- evaluated at the same parameters as they are generated
+- without varying the detected number of bursts
+is generated.
+
+In general, one will want to generate the samples with one set of parameters,
+and then evaluate them with another set of grids.
+Hence the saving of 'mc_sample.npy'
+
+"""
+
+import argparse
+
+import numpy as np
+from zdm import zdm
+#import pcosmic
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+import matplotlib
+from pkg_resources import resource_filename
+import os
+import sys
+
+import scipy as sp
+import time
+from matplotlib.ticker import NullFormatter
+from zdm import iteration as it
+
+from zdm import survey
+from zdm import cosmology as cos
+from zdm import beams
+
+import pickle
+import copy
+
+matplotlib.rcParams['image.interpolation'] = None
+
+defaultsize=14
+ds=4
+font = {'family' : 'normal',
+        'weight' : 'normal',
+        'size'   : defaultsize}
+matplotlib.rc('font', **font)
+
+##### definitions of parameters ########
+	# pset defined as:
+	# [0]:	log10 Emin
+	# [1]:	log10 Emax
+	# [2]:	alpha (spectrum: nu^alpha)
+	# [3]:	gamma
+	# [4]:	sfr n
+	# [5}: log10 mean host DM
+	# [6]: log10 sigma host DM
+	
+from zdm import misc_functions# import *
+#import pcosmic
+
+defaultsize=14
+ds=4
+font = {'family' : 'normal',
+        'weight' : 'normal',
+        'size'   : defaultsize}
+matplotlib.rc('font', **font)
+
+def main():
+
+	############## Initialise cosmology ##############
+	cos.init_dist_measures()
+	
+	# get the grid of p(DM|z). See function for default values.
+	# set new to False once this is already initialised
+	zDMgrid, zvals,dmvals=misc_functions.get_zdm_grid(
+		new=True,plot=False,method='analytic')
+	# NOTE: if this is new, we also need new surveys and grids!
+	
+	# constants of beam method
+	thresh=0
+	method=2
+	
+	
+	# sets which kind of source evolution function is being used
+	source_evolution=0 # SFR^n scaling
+	#source_evolution=1 # (1+z)^(2.7n) scaling
+	
+	
+	# sets the nature of scaling with the 'spectral index' alpha
+	alpha_method=0 # spectral index interpretation: includes k-correction. Slower to update
+	#alpha_method=1 # rate interpretation: extra factor of (1+z)^alpha in source evolution
+	
+	############## Initialise surveys ##############
+	
+	# constants of intrinsic width distribution
+	Wlogmean=1.70267
```

Comment:
For this 'fixed' parameters, we should have a separate `dict` that defines them and carries them around.  I can also take it as my task to do so.

Reply:
In theory the width could be included in the general parameters. From looking at the output, both width, and the host DM properties, are not strongly correlated with source evolution or luminosity function, and I could imagine in an mcmc fitting the whole lot together. So putting this in a dict, then having 'cube' take arguments to loop over parameters as "Emin 40 42 11" or some such where it looks up "Emin" in the dict would make sense. So I would not make a separate dict for this one.